proc/proc_windows.go

```diff
@@ -507,39 +536,65 @@ func (dbp *Process) wait(pid, options int) (int, *sys.WaitStatus, error) {
 }
 
 func (dbp *Process) setCurrentBreakpoints(trapthread *Thread) error {
-	// TODO: In theory, we should also be setting the breakpoints on other
-	// threads that happen to have hit this BP. But doing so leads to periodic
-	// failures in the TestBreakpointsCounts test with hit counts being too high,
-	// which can be traced back to occurrences of multiple threads hitting a BP
-	// at the same time.
+	err := trapthread.SetCurrentBreakpoint()
+	if err != nil {
+		return err
+	}
 
-	// My guess is that Windows will correctly trigger multiple DEBUG_EVENT's
-	// in this case, one for each thread, so we should only handle the BP hit
-	// on the thread that the debugger was evented on.
+	for _, thread := range dbp.Threads {
+		thread.running = false
+		_, err := _SuspendThread(thread.os.hThread)
+		if err != nil {
+			return err
+		}
+	}
 
-	return trapthread.SetCurrentBreakpoint()
+	for {
+		var err error
+		var tid int
+		dbp.execPtraceFunc(func() {
+			err = _ContinueDebugEvent(uint32(dbp.Pid), uint32(dbp.os.breakThread), _DBG_CONTINUE)
```

Comment:
This is the only thing that sticks out to me in this patch set. Here, we're continuing a thread and waiting for a debug event, however in the meantime it's possible that the thread will be executing instructions. This function should not have that side effect.

A non-blocking wait for debug event should suffice, yeah? If there's no events we simply move on. I don't understand the need to continue a thread in a function that is setting state on our thread objects.